### PR TITLE
Implement Pioasm for Go (TinyGo)

### PR DESCRIPTION
**Pull request by soypat**
This PR answers the issue https://github.com/tinygo-org/tinygo/issues/1953, basically answering the question "How PIO will be implemented in TinyGo?".

Most of the discussion of design is also contained inside https://github.com/tinygo-org/tinygo/pull/1983 (initial PR)

In the end development continued in a separate repository: https://github.com/tinygo-org/pio

It'd be great to get a green-light from main TinyGo stakeholders: @deadprogram, @aykevl, @kenbell

### CODE REVIEW

File: tools/pioasm/go_output.cpp [hunk 1]
```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2020 Raspberry Pi (Trading) Ltd.
```
Comment (deadprogram) [inline tools/pioasm/go_output.cpp:2]: This should probably be updated to reflect that it is an external contribution from "The TinyGo Authors". But IANAL to be clear.

**Comment by deadprogram** [comment]
I think this would be a good feature to make available to the broader community now that it is working. :smile_cat: 

Kudos to @soypat for doing all the heavy lifting on this!

**Comment by peterharperuk** [comment]
I would imagine we'd need some "test" code

**Comment by soypat** [comment]
While we wait for objetors to the PR I'd like to ask if there's a predetermined we could add examples on how to use this with Go?
Would it be acceptable to add a link in the pioasm tool help text pointing to examples of Go usage? Namely [piolib](https://github.com/tinygo-org/pio/tree/main/rp2-pio/piolib)

**Comment by deadprogram** [comment]
Whoops, finally getting back to this. :smiley_cat: 

@soypat is there anything still needing to be done on this PR? Also, anything needing to be updated in https://github.com/tinygo-org/pio to match it?

**Comment by soypat** [comment]
@deadprogram  All that is needed is this being merged as the pio library is already dependent on this!

**Review by kilograham — APPROVED** [review]

**Comment by deadprogram** [comment]
Thank you very much for working on this @soypat and to @kilograham for review!

**Comment by nickaxgit** [comment]
Thank you chaps, excited to try this … been watching from the sidelines, PIO in Go should be an awesome combination, much of the Pi Pico Toolchain is extraordinarily complex, and/or broken (at least on my machines, in every flavour).. hoping that TinyGo will be my salvation.

**Pull request merged**